File: Backend/Entity/Order.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Backend.Entity;

[Table("order")]
public class Order
{
    [Column("order_id")]
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int OrderId { get; set; }
    
    public decimal TotalAmount { get; set; }

    [ForeignKey("User")]
    public string UserId { get; set; }

    public User User { get; set; }

    public List<Product> Products { get; set;}
}

File: Frontend/src/app/model/order.ts
// order.model.ts

import { Product } from "./product";

export class order {
    OrderId: number;
    TotalAmount : number;
    UserId : string;
    Products : Product[];
    

    constructor(OrderId: number, TotalAmount : number,
         UserId : string, Products : Product[]) {
      this.OrderId = OrderId;
      this.TotalAmount = TotalAmount;
      this.UserId = UserId;
      this.Products = Products;
    }
  }
  